src/coreclr/nativeaot/System.Private.CoreLib/src/System/EventReporter.cs

```diff
@@ -0,0 +1,194 @@
+            _description.AppendLine(new string((sbyte*)utf8version));
+
+            switch (_eventType)
+            {
+                case RhFailFastReason.UnhandledException:
+                case RhFailFastReason.UnhandledExceptionFromPInvoke:
+                    _description.AppendLine("Description: The process was terminated due to an unhandled exception.");
+                    break;
+                case RhFailFastReason.EnvironmentFailFast:
+                case RhFailFastReason.AssertionFailure:
+                    _description.AppendLine("Description: The application requested process termination through System.Environment.FailFast.");
+                    break;
+                case RhFailFastReason.InternalError:
+                    _description.AppendLine("Description: The process was terminated due to an internal error in the .NET Runtime ");
+                    break;
+                default:
+                    Debug.Fail($"Unknown {nameof(RhFailFastReason)}");
+                    break;
+            }
+        }
+
+        public void AddDescription(string s)
+        {
+            Debug.Assert(_eventType is RhFailFastReason.UnhandledException
+                or RhFailFastReason.EnvironmentFailFast or RhFailFastReason.AssertionFailure
+                or RhFailFastReason.UnhandledExceptionFromPInvoke or RhFailFastReason.InternalError);
+            if (_eventType is RhFailFastReason.EnvironmentFailFast or RhFailFastReason.AssertionFailure)
+            {
+                _description.Append("Message: ");
+            }
+            else if (_eventType == RhFailFastReason.UnhandledException)
+            {
+                _description.Append("Exception Info: ");
+            }
+            _description.AppendLine(s);
+        }
+
+        public void BeginStackTrace()
+        {
+            Debug.Assert(_eventType is RhFailFastReason.UnhandledException
+                or RhFailFastReason.EnvironmentFailFast or RhFailFastReason.AssertionFailure
+                or RhFailFastReason.UnhandledExceptionFromPInvoke);
+            _description.AppendLine("Stack:");
+        }
+
+        public void AddStackTrace(string s)
+        {
+            // The (approx.) maximum size that Vista appears to allow. Post discussion with the OS event log team,
+            // it has been identified that Vista has taken a breaking change in ReportEventW API implementation
+            // without getting it publicly documented.
+            //
+            // An event entry comprises of string to be written and event header information. Prior to Vista,
+            // 32K length strings were allowed and event header size was over it. Vista onwards, the total
+            // permissible length of the string and event header became 32K, resulting in strings becoming
+            // shorter in length. Hence, the change in size.
+            const int MAX_SIZE_EVENTLOG_ENTRY_STRING_WINVISTA = 0x7C62; // decimal 31842
```

Comment:
```suggestion
            // The (approx.) maximum size that EvemtLog appears to allow.
            //
            // An event entry comprises of string to be written and event header information. The total permissible length of the string and event header is 32K.
            const int MAX_SIZE_EVENTLOG_ENTRY_STRING = 0x7C62; // decimal 31842
```
The story about Vista is not relevant anymore. Could you please it fix it in the .cpp version too?